### ISSUE TITLE
Intel RDT does not work with L2 cache

### ISSUE THREAD

**Issue by wpross** [opencontainers/runc#3550]
I have a host with CAT support for L2 cache. I've created two classes in `/sys/fs/resctrl`, `c0` and `c1`:

```bash
$ grep . -r /sys/fs/resctrl/c*
/sys/fs/resctrl/c0/mode:shareable
/sys/fs/resctrl/c0/schemata:L2:0=03;1=03
/sys/fs/resctrl/c0/size:L2:0=262144;1=262144
/sys/fs/resctrl/c0/cpus:0
/sys/fs/resctrl/c1/mode:shareable
/sys/fs/resctrl/c1/schemata:L2:0=0c;1=0c
/sys/fs/resctrl/c1/size:L2:0=262144;1=262144
/sys/fs/resctrl/c1/cpus:0
/sys/fs/resctrl/cpus:f
/sys/fs/resctrl/cpus_list:0-3
```

I've now created a basic example following the description in Section [Creating an OCI Bundle](https://github.com/opencontainers/runc#creating-an-oci-bundle), created the spec with `runc spec` and just added 

```json
"intelRdt": {
    "closID": "c0",
    "l3CacheSchema": "",
    "memBwSchema": ""
}
```

to the `linux` section, then started the container with `sudo runc run rdttest`. Within the container shell I did `sleep 1m` and checked for the PID outside the container with

```bash
$ ps aux | grep sleep
root      469581  0.0  0.0   1320     4 pts/0    S+   18:53   0:00 sleep 1m
```
then checked the `tasks` files for this PID with

```bash 
$ grep . -r /sys/fs/resctrl/* | grep 469581
/sys/fs/resctrl/tasks:469581
```

and found it within the default root group. However, I would expect it to show up in `/sys/fs/resctrl/c0/tasks` instead.

I did exactly the same test on another machine that supports L3 cache for CAT and it worked like expected.

**Comment by wpross** [comment]
I've run the test with the latest release 1.1. 3 in my above test. When I've started debugging the issue and thus compiled from source, it suddenly worked. I guess that one of the following commits solved the issue:

```bash
| * | 2022-07-28 ea0bd782 libct/intelrdt: check if available iff configured (Cory Snider)
| * | 2022-07-28 56daf36b libct/intelrdt: skip remove unless configured (Cory Snider)    
| * | 2022-07-28 c156bde7 libct/intelrdt: elide parsing mountinfo (Cory Snider)          
| * | 2022-07-28 9f107489 libct/intelrdt: skip reading /proc/cpuinfo (Cory Snider)       
| * | 2022-07-28 13674f43 libct/intelrdt: delete IsMBAScEnabled() (Cory Snider)          
```

Thanks @corhere !

**Comment by kolyshkin** [comment]
So, this is still not working in release-1.1 branch. I figured out why and fixed it.

**Comment by kolyshkin** [comment]
Fixed by #3978 in [runc 1.1.9](https://github.com/opencontainers/runc/releases/tag/v1.1.9).
